### PR TITLE
Add support to ssh via ssh-keys

### PR DESCRIPTION
**Pull request by adesh-rao**
This change adds support to create a cluster with ssh using ssh-keys.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,8 +51,14 @@ activeDirectory:
 clusterCredentials:
   clusterLoginUsername: <YOU_USER_NAME> # Ambari username
   clusterLoginPassword: <YOUR_PASSWORD> # Ambari password
-  sshUsername: <SSH_USER> # SSH username
-  sshPassword: <SSH_PASSWORD> # SSH password
+  sshCredentials:
```
Comment (guptanikhil007) [inline README.md:54]: For better clarification split into two

File: src/main/java/com/microsoft/hdinsight/utils/AzureUtils.java [hunk 2]
```diff
@@ -199,12 +210,46 @@ private static StorageAccount fetchStorageAccount(HWCClusterConfig clusterConfig
...
   private static OsProfile addSSHCredentials(HWCClusterConfig clusterConfig) {
     LOG.info("Setting up SSH credentials...");
+    if (clusterConfig.getClusterCredentials().getSshCredentials() instanceof SSHWithPassword) {
+      return addSSHWithPasswordCredentials((SSHWithPassword) clusterConfig.getClusterCredentials().getSshCredentials());
+    } else {
+      return addSSHWithKeysCredentials((SSHWithKeys) clusterConfig.getClusterCredentials().getSshCredentials());
+    }
+  }
+
+  private static OsProfile addSSHWithKeysCredentials(SSHWithKeys sshWithKeys) {
```
Comment (guptanikhil007) [inline src/main/java/com/microsoft/hdinsight/utils/AzureUtils.java:230]: Test what happens for invalid ssh key

File: README.md (hunk 1 above)
Comment (adesh-rao) [reply README.md:54]: Done.

File: src/main/java/com/microsoft/hdinsight/utils/AzureUtils.java (hunk 2 above)
Comment (adesh-rao) [reply src/main/java/com/microsoft/hdinsight/utils/AzureUtils.java:230]: Modified the code to throw an exception if the ssh key path is non-existent or a directory

**Review by guptanikhil007 — APPROVED** [review]

**Pull request merged**